### DISCUSSION TITLE
.NET5 Core App locks with Quartz integration

### DISCUSSION THREAD

**Discussion by Will-Bill** [quartznet/quartznet#1608] [Q&A]
**Describe the bug**
Occasionally we have our web apps go offline due to a locking issue. Looking at an analysed memory dump, we see:

`1 threads are performing asynchronous work in method 'Quartz.dll!Quartz.Impl.AdoJobStore.ConnectionAndTransactionHolder.Commit(bool)', but are waiting on a synchronous call to method 'System.Private.CoreLib.dll!System.Threading.ManualResetEventSlim.Wait(int, System.Threading.CancellationToken)'. This may cause thread pool starvation and hangs.`

**Version used**

3.4.0

**To Reproduce**

Unable to reproduce outside Azure hosting environment.

**Additional context**
Here is our configuration:

```
  "Quartz": {
    "quartz.plugin.timeZoneConverter.type": "Quartz.Plugin.TimeZoneConverter.TimeZoneConverterPlugin, Quartz.Plugins.TimeZoneConverter"
  },


services.Configure<QuartzOptions>(Configuration.GetSection("Quartz"));
                services.AddQuartz(q =>
                {
                    q.UseMicrosoftDependencyInjectionScopedJobFactory();

                    q.UsePersistentStore(options =>
                    {
                        options.UseSqlServer(providerOptions =>
                        {
                            providerOptions.ConnectionString = Configuration.GetConnectionString("dbCon");
                        });
                        options.UseClustering();
                        options.UseJsonSerializer();
                    });
                });
                services.AddQuartzHostedService(options => { options.WaitForJobsToComplete = true; });
```

The only thing of note is I see we are using a deprecated service registration - UseMicrosoftDependencyInjectionScopedJobFactory().

We will update this and try again, also updating to .NET6.

In the mean time are there any other suggestions / obvious issues with this registration?

**Comment by lahma** [comment]
This could be pretty much anything... maybe you need to check SQL server locks and what could be deadlocked. 

Any custom SQL scripts accessing Quartz tables, any instance configured without clustering and pointing to this clustered setup.

**Comment by wangliguokaka** [comment]
@Will-Bill 
for the description 
"1 threads are performing asynchronous work in method 'Quartz.dll!Quartz.Impl.AdoJobStore.ConnectionAndTransactionHolder.Commit(bool)', but are waiting on a synchronous call to method 'System.Private.CoreLib.dll!System.Threading.ManualResetEventSlim.Wait(int, System.Threading.CancellationToken)'. This may cause thread pool starvation and hangs."

can you tell me that what tool do you use to analyse the memory dump file, thank you.

**Reply by Will-Bill** [reply]
@wangliguokaka i can't remember now but either Azures built in memory analysis or dotMemory

**Reply by wangliguokaka** [reply]
OK，thank you，did you solve your problem for the quartz？
